src/sage/dynamics/arithmetic_dynamics/projective_ds.py

```diff
@@ -8437,6 +8437,14 @@ def normal_form(self, return_conjugation=False):
               To:   Finite Field in z2 of size 3^2
               Defn: 1 |--> 1
 
+        Fixes :issue:`38012` by not forcing univariate polynomial to be univariate::
+
+            sage: R.<z> = PolynomialRing(QQ)
```

Comment:
The comment says an implementation detail that is not understandable by a user. The usual way to introduce a test (example) for a fix is
```
TEST:

Check that :issue:`38012` is fixed::

    sage: R.<z> = PolynomialRing(QQ)

```